test/Microsoft.VisualStudio.Threading.Tests/JoinableTaskContextTests.cs

```diff
@@ -45,6 +45,41 @@ public void IsWithinJoinableTask()
         });
     }
 
+    [Fact]
+    public void IsMainThreadBlockedByAnyoneTrue()
+    {
+        Assert.False(this.Context.IsMainThreadBlockedByAnyone);
+        this.Factory.RunAsync(async delegate
+        {
+            await TaskScheduler.Default.SwitchTo(alwaysYield: true);
+
+            Assert.False(this.Context.IsMainThreadBlockedByAnyone);
+
+            await this.Factory.SwitchToMainThreadAsync();
+            this.Factory.Run(async delegate
+            {
+                Assert.True(this.Context.IsMainThreadBlockedByAnyone);
```

Comment:
I'm confused as to why this should be `true` when the main thread isn't blocked. You're executing this code on the main thread, so by definition the main thread isn't blocked. I'm guessing this throws a failure right now, but per my comment above, your test isn't failing as a result of it.

Reply:
Test updated